thorlcr/master/thgraphmanager.cpp

```diff
@@ -60,10 +61,145 @@ class CJobManager : public CSimpleInterface, implements IJobManager, implements
     StringAttr          currentWuid;
     ILogMsgHandler *logHandler;
 
+    CJobMaster *getCurrentJob() { CriticalBlock b(jobCrit); return jobs.ordinality() ? &OLINK(jobs.item(0)) : NULL; }
     bool executeGraph(IConstWorkUnit &workunit, const char *graphName, const SocketEndpoint &agentEp);
     void addJob(CJobMaster &job) { CriticalBlock b(jobCrit); jobs.append(job); }
     void removeJob(CJobMaster &job) { CriticalBlock b(jobCrit); jobs.zap(job); }
 
+    class CThorDebugListener : public CSimpleInterfaceOf<IInterface>, implements IThreaded
+    {
+    protected:
+        CThreaded threaded;
+        unsigned port;
+        Owned<ISocket> sock;
+        CJobManager &mgr;
+    private:
+        volatile bool running;
+    public:
+        CThorDebugListener(CJobManager &_mgr) : threaded("CThorDebugListener", this), mgr(_mgr)
+        {
+            port = globals->getPropInt("DebugPort", 9877);
+            running = true;
+            threaded.start();
+        }
+        ~CThorDebugListener()
+        {
+            running = false;
+            if (sock)
+                sock->cancel_accept();
+            threaded.join();
+        }
+        virtual void main()
+        {
+            sock.setown(ISocket::create(port));
+            while (running)
+            {
+                try
+                {
+                    ISocket *client = sock->accept(true);
+                    if (client)
+                    {
+                        client->set_linger(-1);
+                        CSafeSocket ssock(client);
+                        StringBuffer rawText;
+                        unsigned priority = (unsigned) -2;
+                        unsigned memused = 0;
+                        IpAddress peer;
+                        bool continuationNeeded;
+                        bool isStatus;
+                        ssock.querySocket()->getPeerAddress(peer);
+                        DBGLOG("Reading debug command from socket...");
+                        if (!ssock.readBlock(rawText, WAIT_FOREVER, NULL, continuationNeeded, isStatus, 1024*1024))
+                        {
+                            if (traceLevel > 8)
```

Comment:
Where's 'traceLevel' coming from? Hm, looks like external of roxiehelper.hpp.
Shouldn't use that.
Thor doesn't really have a traceLevel, though it should, but separate issue.

For now I'd just remove this condition and always trace this message.